### PR TITLE
RHDEVDOCS-5704: Updating install build strategies section

### PR DESCRIPTION
**Pull request by shipsing**
**Purpose**: To resolve this issue: https://issues.redhat.com/browse/RHDEVDOCS-5704

**Aligned team**: DevTools

**Version for cherrypicking**: build-docs-1.0

**Content for preview**: 

- Added the conceptual information in a separate assembly called Build strategies and removed installing build startegies assembly to avoid any confusion: https://67283--docspreview.netlify.app/openshift-builds/latest/about/build-strategies
- Added the Operator behavior (The Builds for Red Hat OpenShift Operator automatically installs these strategies for use.) in the RN as well: https://67283--docspreview.netlify.app/openshift-builds/latest/about/ob-release-notes#new-features-1-0_ob-release-notes

**SME review**:  @apoorvajagtap @jkhelil 

**QE review**:  @jitendar-singh 

**Peer review**: @lpettyjo

### CODE REVIEW

**Comment by shipsing** [comment]
@jkhelil @apoorvajagtap Do we have any impact on the [Uninstalling Builds section](https://67283--docspreview.netlify.app/openshift-builds/latest/uninstalling/uninstalling-openshift-builds) as well? When we uninstall the Operator, would the build strategies also be removed alongside automatically? or Any other specific behavior to highlight?

File: about/build-strategies.adoc [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
+= Build strategies
 include::_attributes/common-attributes.adoc[]
-:context: installing-sample-build-strategies
+:context: build-strategies
 
 toc::[]
 
-You can install a build strategy or cluster build strategy on the {ocp-product-title} cluster. A `ClusterBuildStrategy` resource is available cluster-wide and a `BuildStrategy` resource is available within a namespace.
-
+You can use a curated set of build strategies or cluster build strategies on the {ocp-product-title} cluster. The {builds-operator} automatically installs these strategies for use. This automated installation of build strategies helps you to quickly get started with {builds-shortname}.
```
Comment (apoorvajagtap) [inline about/build-strategies.adoc:9]:
```suggestion
You can use a curated set of build strategies or cluster build strategies on the {ocp-product-title} cluster. The {builds-operator} automatically installs these strategies for use. This automated installation of cluster build strategies helps you to quickly get started with {builds-shortname}.
```
As we'll have the strategies installed at cluster-scope by operator, I think it'll be better to specify the cluster build strategies here. WDYT?

**Comment by apoorvajagtap** [comment]
@shipsing I don't see any impact on the Uninstallation section. As we are already suggesting to remove the CRDs [here](https://67283--docspreview.netlify.app/openshift-builds/latest/uninstalling/uninstalling-openshift-builds), the removal of CBS CRD will delete the strategies as well.

**Review by jkhelil — APPROVED** [review]

**Review by lpettyjo — COMMENTED** [review]
LGTM!

File: modules/ob-about-buildah.adoc [hunk 2]
```diff
@@ -1,10 +1,37 @@
 // This module is included in the following assembly:
 //
-// * installing/installing-sample-build-strategies.adoc
+// * about/build-strategies.adoc
 
-:_content-type: CONCEPT
+:_content-type: REFERENCE
```
Comment (mjpytlak) [inline modules/ob-about-buildah.adoc:5]:
```suggestion
:_mod-docs-content-type: REFERENCE
```

Comment (mjpytlak) [reply modules/ob-about-buildah.adoc:5]: As part of our adoption of the IBM tool chain (WFM), we are being asked to use `:_mod-docs-content-type:`. Seeing as you are here, please update.

File: modules/ob-about-buildah.adoc [hunk 3]
```diff
@@ -1,10 +1,37 @@
...
+| `registries-search` | array | List of registries to search short name images | ["registry.redhat.io", "quay.io"]
+
+| `dockerfile` | string | Path of the Dockerfile that is used during the build | "Dockerfile"
+
+| `storage-driver` | string | Storage drivers that are used by `buildah`, such as overlay or vfs | "vfs"
+|===
+
+[NOTE]
+====
+For more details, see _Configuring build strategies_ in the _Additional resources_ section.
```
Comment (mjpytlak) [inline modules/ob-about-buildah.adoc:36]:
```suggestion
For more information, see _Configuring build strategies_ in the _Additional resources_ section.
```
I know it is a nit, but our guidelines require "information" [1]
[1] https://redhat-documentation.github.io/supplementary-style-guide/#cross-references

File: modules/ob-about-s2i.adoc [hunk 4]
```diff
@@ -1,9 +1,30 @@
 // This module is included in the following assembly:
 //
-// * installing/installing-sample-build-strategies.adoc
+// * about/build-strategies.adoc
 
-:_content-type: CONCEPT
+:_content-type: REFERENCE
```
Comment (mjpytlak) [inline modules/ob-about-s2i.adoc:5]:
```suggestion
:_mod-docs-content-type: REFERENCE
```

**Review by mjpytlak — COMMENTED** [review]
@shipsing Found a handful of minor items as part of the merge review. PTAL.

File: modules/ob-about-buildah.adoc (hunk 2 above)
Comment (shipsing) [reply modules/ob-about-buildah.adoc:5]: @mjpytlak, I incorporated this suggestion globally and updated content type to  :_mod-docs-content-type: in all modules of this project.

**Comment by shipsing** [comment]
@mjpytlak, I have incorporated all your suggestions. The comment related to changing content type to :_mod-docs-content-type: has been applied globally. Please verify. Thanks

**Review by mjpytlak — APPROVED** [review]

**Pull request merged**